torch/csrc/api/include/torch/nn/module.h

```diff
@@ -547,6 +547,8 @@ class TORCH_API Module : public std::enable_shared_from_this<Module> {
   template <typename ModuleType, typename... ArgumentTypes>
   friend struct AnyModuleHolder;
 
+  friend class ParameterDictImpl;
```

Comment:
That makes more sense to me! I am avoiding changing the `module.h`, so instead of changing the access of `parameters_`, I add the friend class instead.